### DISCUSSION TITLE
Hi guys, i need some advice.

### DISCUSSION THREAD

**Discussion by dos2803** [community/community#141325] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi, I'm a banking student but I want to learn programming can I do any job if I don't have a degree in Programming?

**Comment by ghost** [comment]
Yes, you will be able to get a job without a degree @dos2803, but you will have to learn on your own and get some certifications.

Here are some resources that will help you get started and learn.

https://github.com/EbookFoundation/free-programming-books
https://www.freecodecamp.org/

**Reply by dos2803** [reply]
Thank you, but can you give me some certification I need for back-end Java development? I'm going to learn Java.

**Comment by Ahsankhalid618** [comment, marked as answer]
@dos2803 Trust me, you don't need a degree to learn programming, You can learn any programming skills online either on YouTube or by course. My purest advice for you is to start your journey with Web development as it is easy to learn and it will give you a clear roadmap for other fields like Mobile dev etc. Just practice what you learn and within 6 - 8 months, you will be a much better programmer than most of the university graduates.

### If you like my answer, Kindly Accept it as an answer. thanks ❤️

**Reply by kalix127** [reply]
you're absolutely right, you pretty much described me :)

**Reply by Ahsankhalid618** [reply]
Thanks 😊

**Reply by dos2803** [reply]
Should I learn back-end Java? Is it still popular to get jobs?
Do I need some certifications if I'm not a university graduate?

**Reply by Ahsankhalid618** [reply]
Certification is not necessary, nor they are difficult to get, Just learn the skills and make projects, and they will reflect your worth. As for java, yes, if you want to learn java, you should also learn java frameworks like springboot, which is quite popular now adays.

**Reply by dos2803** [reply]
Hmmm about a project where can I find and make that if I'm done with study?

**Reply by Ahsankhalid618** [reply]
> Hmmm about a project where can I find and make that if I'm done with study?

you will need a code editor like VSCode where you will write code to create projects, once you are done with your studies, you will start doing projects in the code editor, 

### If I have been helpful to you, Kindly Accept this thread as an answer, it will be beneficial for everyone.❤️

**Reply by dos2803** [reply]
Can you accept me on Facebook?

**Reply by Ahsankhalid618** [reply]
> Can you accept me on Facebook?

I don't use Facebook anymore, Connect with me on Linkedin.

**Reply by dos2803** [reply]
Ahh oke.

**Comment by d-coder111** [comment]
Absolutely @dos2803 ! 🎉 You can pursue a programming career even without a formal degree in computer science. Here are some steps to get started:

1. **Self-Taught Learning**: 📚 There are countless online resources, tutorials, and courses (like Codecademy, freeCodeCamp, Coursera) to help you learn programming from scratch. No degree needed!
2. **Build a Portfolio**: 💻 As you learn, start working on projects and upload them to GitHub. A strong portfolio showcasing your skills is often more valuable than a degree. 
3. **Freelancing**: 💼 Once you're confident in your skills, freelancing on platforms like Upwork, Fiverr, or Freelancer can help you gain practical experience and build a reputation.
4. **Certifications**: 🏅 Earning certifications from platforms like Google, AWS, or Microsoft can further validate your skills. These are widely recognized by employers and can make your resume stand out.
5. **Open Source Contributions**: 🤝 Contributing to open-source projects on GitHub is a great way to gain real-world experience, collaborate with other developers, and showcase your skills to potential employers.
6. **Networking**: 🌐 Join tech meetups, connect on LinkedIn, and attend hackathons. Networking can help you find opportunities, internships, and jobs—even if you're new to the field.


Pick a language like Python 🐍 or JavaScript 📜, and stay consistent with your learning! There are plenty of success stories of people landing programming jobs without a traditional degree. You got this! 🚀

Feel free to mark this as the answer if it helped! ✅

**Reply by dos2803** [reply]
Thanks for your detailed information.

**Comment by iftekharalammithu** [comment]
Yes! 
Anyone can pursue a career in programming without a degree. Before that, you must be an expert in programming.

**Reply by dos2803** [reply]
Thank you, but how can I be an expert if I don't have any guidance from school, to make some project or something like that?

**Comment by Arx-F** [comment]
Yes, you can absolutely pursue a job in programming even without a formal degree in the field. Many companies are open to hiring self-taught developers or those from non-traditional backgrounds, as long as they have the necessary skills.

Here are a few tips to help you get started:

### Steps to Start:
1. **Learn the Basics**: Start with beginner-friendly languages like **[Python](https://www.python.org/)** or **[JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)**. Master key concepts like **data structures**, **algorithms**, and **problem-solving**.
2. **Build Projects**: Create real-world projects and upload them to **[GitHub](https://github.com/)** to showcase your skills.
3. **Certifications**: Use platforms like **[Coursera](https://www.coursera.org/)**, **[Udemy](https://www.udemy.com/)**, or **[freeCodeCamp](https://www.freecodecamp.org/)** to earn certifications.
4. **Freelancing**: Start small on platforms like **[Upwork](https://www.upwork.com/)** or **[Fiverr](https://www.fiverr.com/)** to gain experience.
5. **Open Source Contributions**: Contribute to **[GitHub](https://github.com/)** projects to gain practical knowledge and network with other developers.

### Coding Practice Websites:
- **[LeetCode](https://leetcode.com/)**, **[HackerRank](https://www.hackerrank.com/)**, **[Codewars](https://www.codewars.com/)**, **[CodeSignal](https://codesignal.com/)**, **[Codeforces](https://codeforces.com/)** (for algorithms & interview prep).
- **[freeCodeCamp](https://www.freecodecamp.org/)**, **[Codecademy](https://www.codecademy.com/)**, **[Edabit](https://edabit.com/)** (for structured learning and projects).

With consistent learning and building your portfolio, you can land a job in programming even without a formal degree!
